torch/onnx/__init__.py

```diff
@@ -224,13 +232,38 @@ def _run_symbolic_function(g, n, inputs):
             op_name = n.kind()[:-1]
         else:
             op_name = n.kind()
-        if not hasattr(torch.onnx.symbolic, op_name):
-            warnings.warn("ONNX export failed on {} because torch.onnx.symbolic.{} does not exist"
-                          .format(op_name, op_name))
-            return None
-        fn = getattr(torch.onnx.symbolic, op_name)
-        attrs = {k: n[k] for k in n.attributeNames()}
-        return fn(g, *inputs, **attrs)
+        if globals()['aten']:
+            attrs = {}
+            def type_suffix(var):
+                if isinstance(var, numbers.Integral):
+                    return "i"
+                elif isinstance(var, float):
+                    return "f"
+                elif isinstance(var, string_classes):
+                    return "s"
+                elif torch.is_tensor(var):
+                    return "t"
+                else:
+                    raise ValueError("Unknown attribute type when exporting ATen ops:".format(type(var)))
+            for k in n.attributeNames():
+                val = n[k]
+                if not isinstance(val, string_classes) and not torch.is_tensor(val) and isinstance(val, collections.Iterable):
+                    attrs[k + "_" + type_suffix(val[0])] = val
+                else:
+                    attrs[k + "_" + type_suffix(val)] = val
+            node = _graph_at(g, op_name, *inputs, **attrs)
+            # Only max_pool2d return two values
+            if op_name == "max_pool2d":
```

Comment:
This is a good point. Actually, this is a good opportunity to extend the `run_symbolic_function` so that other multiple return ATen functions get handled properly. Basically, what we should do is pass the number of expected outputs based on the trace to the function here, so we can consequently return the correct number of outputs.

Suppose that there are two expected outputs. IIUC, the ATen op in Caffe2 will actually return two outputs. So really, we should create a multi-return ATen op, not a single return one with None in the second parameter. Though, I could be wrong (having not tested this case.) Do you know?

Reply:
Just confirmed that with @zdevito, yes, it will also output two tensors in Caffe2. Yeah, let's handle multiple output ATen ops properly.